libraries/rush-lib/src/cli/scriptActions/PhasedScriptAction.ts

```diff
@@ -459,6 +460,42 @@ export class PhasedScriptAction extends BaseScriptAction<IPhasedCommandConfig> {
     await this._executeOperations(initialOptions);
   }
 
+  private _registerWatchModeInterface(projectWatcher: ProjectWatcher): void {
+    const toggleWatcherKey: string = 'w';
+    const buildOnceKey: string = 'b';
+
+    const terminal: ITerminal = this._terminal;
+
+    process.stdin.setRawMode(true);
+    process.stdin.resume();
+    process.stdin.setEncoding('utf8');
+    process.stdin.on('data', (key: string) => {
+      switch (key) {
+        case toggleWatcherKey:
+          if (projectWatcher.isPaused) {
+            terminal.writeLine(`Resuming project watcher...`);
+            projectWatcher.resume();
+          } else {
+            terminal.writeLine(`Pausing project watcher...`);
+            projectWatcher.pause();
+          }
+          break;
+        case buildOnceKey:
+          if (projectWatcher.isPaused) {
+            terminal.writeLine(`Building once...`);
+            projectWatcher.resume();
+            setTimeout(() => {
+              projectWatcher.pause();
+            }, 1000);
```

Comment:
You should be able to get away with doing this immediately and not need a timeout.